### PR TITLE
#2741 - Ministry Filter: Search Programs & Offerings [Data Table Conversion to Vuetify]

### PR DESCRIPTION
**Pull request by sh16011993**
### As a part of this PR, the following task was completed:

- Converted Prime Vue Table to Vuetify Data Table.

### Screenshots:

**Added the loader while the data table loads:**

image

----------------------------------------------------------------------------------------------------------------------------------

**Data table with data:**

image

----------------------------------------------------------------------------------------------------------------------------------

**Data table without any records:**

image

### CODE REVIEW

File: sources/packages/web/src/types/contracts/DataTableContract.ts [hunk 1]
```diff
@@ -393,3 +393,14 @@ export const ApplicationRestrictionManagementHeaders = [
     key: "removeBypassRule",
   },
 ];
+/**
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/contracts/DataTableContract.ts:396]: Please add a blank line following the file pattern.

File: sources/packages/web/src/types/contracts/DataTableContract.ts [hunk 2]
```diff
@@ -393,3 +393,14 @@ export const ApplicationRestrictionManagementHeaders = [
...
+ * Program Headers.
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/contracts/DataTableContract.ts:397]: "Program headers."

File: sources/packages/web/src/views/aest/institution/Programs.vue [hunk 3]
```diff
@@ -21,77 +20,44 @@
...
-          :paginator="true"
-          :rows="DEFAULT_PAGE_LIMIT"
-          :rowsPerPageOptions="PAGINATION_LIST"
-          :totalRecords="institutionProgramsSummary.count"
-          @page="pageSortEvent($event)"
-          @sort="pageSortEvent($event)"
-          :loading="loading"
+        <toggle-content
+          :toggled="!loading && !institutionProgramsSummary.count"
+          message="No records found."
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Programs.vue:25]: Unless a requirement please remove the message to display the default one.

File: sources/packages/web/src/views/aest/institution/Programs.vue [hunk 4]
```diff
@@ -21,77 +20,44 @@
...
-              ></status-chip-program> </template
-          ></Column>
-          <Column header="Action">
-            <template #body="slotProps">
+                :status="item.programStatus"
+                :is-active="item.isActive && !item.isExpired"
+              ></status-chip-program>
+            </template>
+            <template #[`item.action`]="{ item }">
               <v-btn
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Programs.vue:53]: This button is missing the color configuration. Can you please add the below?
```
color="primary"
```

File: sources/packages/web/src/views/aest/institution/Programs.vue [hunk 5]
```diff
@@ -175,17 +141,6 @@ export default defineComponent({
         },
       });
     };
-    const pageSortEvent = async (event: any) => {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Programs.vue:178]: As we talked, let's try to have the server-side pagination included in this PR also.

**Review by andrewsignori-aot — COMMENTED** [review]
Nice start and thanks for breaking the PR. Please take a look at the comments.

File: sources/packages/web/src/views/aest/institution/Programs.vue [hunk 6]
```diff
@@ -21,77 +20,46 @@
...
+          :toggled="!loading && !institutionProgramsSummary.count"
         >
-          <template #empty>
-            <p class="text-center font-weight-bold">No records found.</p>
-          </template>
-          <Column
-            :field="ProgramSummaryFields.SubmittedDate"
-            header="Date Submitted"
-            :sortable="true"
+          <v-data-table
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Programs.vue:26]: The pagination is not working for me. Am I missing something?
The below programs page from institutions can be used as a reference. Can you please take a look?
https://github.com/bcgov/SIMS/blob/89d4bde5c5499a01afc2707385e90064ccc8cf86/sources/packages/web/src/views/institution/locations/programs/LocationPrograms.vue#L40

One main difference is the use the `v-data-table` vs `v-data-table-server`.

**Review by andrewsignori-aot — COMMENTED** [review]
Thanks for making the changes. Can you check if the server validation is working and review the below comment?
https://github.com/bcgov/SIMS/pull/3805#discussion_r1810974740

File: sources/packages/web/src/views/aest/institution/Programs.vue [hunk 7]
```diff
@@ -128,16 +98,17 @@ export default defineComponent({
...
+    const currentPage = ref();
+    const currentPageLimit = ref();
+    const loading = ref(true);
 
     const getProgramsSummaryList = async (
       institutionId: number,
-      rowsPerPage: number,
-      page: number,
+      rowsPerPage = DEFAULT_PAGE_LIMIT,
+      page = DEFAULT_DATATABLE_PAGE_NUMBER,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/aest/institution/Programs.vue:108]: Please revert this twi line to avoid the Sonarcloud issue.

Comment (sh16011993) [reply sources/packages/web/src/views/aest/institution/Programs.vue]: I made those 2 as the last parameters. Does that work?

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for making the changes, looks good 👍

**Review by andrepestana-aot — APPROVED** [review]
Looks good! Great job!

**Pull request merged**